bundles/org.openhab.binding.chatgpt/src/main/java/org/openhab/binding/chatgpt/internal/ChatGPTHLIService.java

```diff
@@ -0,0 +1,438 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.chatgpt.internal;
+
+import static org.openhab.binding.chatgpt.internal.ChatGPTHLIConstants.SERVICE_ID;
+import static org.openhab.binding.chatgpt.internal.ChatGPTHLIConstants.SERVICE_PID;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.chatgpt.internal.dto.ChatFunction;
+import org.openhab.binding.chatgpt.internal.dto.ChatFunctionCall;
+import org.openhab.binding.chatgpt.internal.dto.ChatMessage;
+import org.openhab.binding.chatgpt.internal.dto.ChatRequestBody;
+import org.openhab.binding.chatgpt.internal.dto.ChatResponse;
+import org.openhab.binding.chatgpt.internal.dto.ChatToolCalls;
+import org.openhab.binding.chatgpt.internal.dto.ChatTools;
+import org.openhab.binding.chatgpt.internal.dto.ToolChoice;
+import org.openhab.binding.chatgpt.internal.dto.functions.ItemsControl;
+import org.openhab.core.config.core.ConfigurableService;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemNotFoundException;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.items.events.ItemEventFactory;
+import org.openhab.core.library.items.RollershutterItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.UpDownType;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.TypeParser;
+import org.openhab.core.voice.text.HumanLanguageInterpreter;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Artur Fedjukevits - Initial contribution
+ */
+@Component(scope = ServiceScope.PROTOTYPE, configurationPid = SERVICE_PID, property = Constants.SERVICE_PID + "="
```

Comment:
Why `ServiceScope.PROTOTYPE` here?

Reply:
Just copied from your example: [ChatGPTModelOptionProvider](https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.chatgpt/src/main/java/org/openhab/binding/chatgpt/internal/ChatGPTModelOptionProvider.java) :) Sorry, I didn't even figure out what it was for.